### ISSUE TITLE
Error when using regularizers, but model does not have layer

### ISSUE THREAD

**Issue by mctigger** [ncullen93/torchsample#33]
I get the following error when my model does not have any convolutional layers, but sub-models do.

> torchsample/modules/module_trainer.py", line 595, in fit_loader
    batch_logs['regularizer_loss'] = regularizer_loss.data[0]

When I add 
`self.conv1 = nn.Conv2d(64, 32, kernel_size=3, stride=1, padding=1)`

it works. (Though I do not use self.conv1 for calculations).

Are regularizers for nested models even supported? :/

**Comment by ncullen93** [comment]
interesting.. No the regularizers only search for the first layer of modules (e.g. `module.named_modules()` . This is something I will definitely try to implement. thanks
